feat(profiling): Do not disable allocation profiling on fixed PHP version with active JIT

realFlowControl opened the pull request:
### Description

In #2088 we disabled allocation profiling on PHP Version >= 8.0.0 when an active JIT is detected. This PR will incorporate the fact that PHP 8.1.21 and 8.2.8 have a fix for this bug, so we will now disable allocation profiling only when JIT is active on PHP 8.0, 8.1.0 - 8.1.20 and 8.2.0 - 8.2.7.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

realFlowControl commented on profiling/src/lib.rs:
```diff
@@ -83,6 +83,21 @@ lazy_static! {
             .expect("Reflection's zend_module_entry to be found and contain a valid string")
     };
 
+    static ref PHP_VERSION_STRUCTURED: Option<(u32, u32, u32)> = {
+        let mut iter = PHP_VERSION.split('.').fuse();
+
+        let major = iter.next()?.parse().ok()?;
+        let minor = iter.next()?.parse().ok()?;
+        let patch = iter
+            .next()?
+            .split(|c: char| !c.is_numeric())
+            .next()?
+            .parse()
+            .ok()?;
+
+        Some((major, minor, patch))
+    };
+
```
I am wondering if we should try and go for something `PHP_VERSION_ID` like, so that we can just call `php_version_id()` from PHP 8.3 onwards?
https://github.com/php/php-src/pull/11875

realFlowControl replied:
Ah yes, this looks nicer 😄

morrisonlevi reviewed (commented):
There should already be a `PHP_VERSION_ID` constant in the bindings, e.g. `zend::PHP_VERSION_ID`. It's the version at compile-time, but of course this simplified certain parts of the PR.

realFlowControl commented:
You are right about that, but in this case I need the `PHP_VERSION_ID` at runtime as we are building for example on PHP 8.2.8 the Profiler version for PHP 8.2. Someone might use it with PHP 8.2.7 which does not have the JIT patch applied.

morrisonlevi reviewed (approved):
Approved from me, but please test my changes out before merging.

realFlowControl commented:
Clippy complained with https://rust-lang.github.io/rust-clippy/master/index.html#/manual_range_contains
I "fixed" this, added a test and fixed the PHP 8.3 branch. Thanks for approving!
